test/e2e/testdata/artifactgc/artgc-has-gc-but-failed-no-artifacts.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: artifact-gc-but-failed-no-artifacts-
```

Comment:
the filename should match this by convention

also I think we should have this named similarly to `artgc-artifact-not-written.yaml` from #13066; the only difference is that that one has another successful artifact. (also `has-gc` is redundant here)

I might name this one `artgc-artifact-not-written-failed`? Or maybe this one as `artgc-artifact-not-written` and rename that one to `artgc-artifact-not-written-and-written` or something